### PR TITLE
ENH: Add type parameter to literals

### LINKED ISSUE #979: Add argument to ibis.literal to allow a specific type for the literal
Something like:

```python
import ibis
ibis.literal(2, type='int64')
```

### PR DESCRIPTION
**Pull request by cpcloud**
Closes #979

- [x] tests

### CODE REVIEW

**Comment by cpcloud** [comment]
This needs tests.

**Comment by cpcloud** [comment]
@wesm Can you review this when you get a chance?

**Comment by wesm** [comment]
Looking

File: ibis/expr/types.py [hunk 1]
```diff
@@ -1243,22 +1241,54 @@ def as_value_expr(val):
...
+    >>> x = ibis.literal(42)
+    >>> x.type()
+    int8
+    >>> y = ibis.literal(42, type='double')
+    >>> y.type()
+    double
+    >>> ibis.literal('foobar', type='int64')  # doctest: +ELLIPSIS
+    Traceback (most recent call last):
+      ...
+    TypeError: Value 'foobar' cannot be safely coerced to int64
```
Comment (wesm) [inline ibis/expr/types.py]: Ha oh man. Doctests

**Pull request closed without merging**